global_helpers/panther_greynoise_helpers.py

```diff
@@ -0,0 +1,218 @@
+# pylint: disable=too-many-public-methods
+import datetime
+
+from dateutil import parser
+from panther_base_helpers import deep_get
+
+
+class PantherGreyNoiseException(Exception):
+    def __init__(self, sublevel):
+        if sublevel == "advanced":
+            message = (
+                "This account is configured with an advanced GreyNoise Subscription. Please "
+                "use GreyNoiseAdvanced and GreyNoiseRIOTAdvanced "
+            )
+        elif sublevel == "basic":
+            message = (
+                "This account is configured with a basic GreyNoise Subscription. Please use "
+                "GreyNoiseBasic and GreyNoiseRIOTBasic "
+            )
+        else:
+            message = "Unknown Error Reading GreyNoise Data"
+        super().__init__(message)
+
+
+class GreyNoiseBasic:
+    def __init__(self, event):
+        self.noise = deep_get(event, "p_enrichment", "greynoise_noise_basic")
+        self.sublevel = "basic"
+
+    def __getattr__(self, name):
+        def advanced_only():
+            advanced_methods = [
+                method
+                for method in dir(GreyNoiseAdvanced)
+                if method.startswith("__") is False and method not in dir(self)
+            ]
+            if name in advanced_methods:
+                raise PantherGreyNoiseException(self.sublevel)
```

Comment:
So I think the combination of this code block and and these [ helper functions ](https://github.com/panther-labs/panther-analysis/pull/377/files#diff-7cec10bf13c890c56d7bdb3b9dfe092938baa86d2d61c2b495b11fbefa787610R208-R218) should completely abstract the selection of the appropriate GreyNoise Class function from the user. 

The two helpers below will select the correct GreyNoise (RIOT) data sets based on what is present in the event.  
The code above will raise an exception only when an advanced method is called on a basic object.  We will still need to be diligent about the detections we ship not to blindly assume an advanced subscription but I think thats a reasonable practice regardless of the helper design. This will still need some live data testing, but I think this should do the trick

cc @lindsey-w